src/mlpack/tests/main_tests/load_save_image_test.cpp

```diff
@@ -0,0 +1,165 @@
+/**
+ * @file load_save_image_test.cpp
+ * @author Jeffin Sam
+ *
+ * Test mlpackMain() of load_save_image_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#define BINDING_TYPE BINDING_TYPE_TEST
+
+#include <mlpack/core.hpp>
+static const std::string testName = "LoadSaveImage";
+
+#include <mlpack/core/util/mlpack_main.hpp>
+#include <mlpack/methods/preprocess/load_save_image_main.cpp>
+
+#include "test_helper.hpp"
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct LoadSaveImageTestFixture
+{
+ public:
+  LoadSaveImageTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+
+  ~LoadSaveImageTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(LoadSaveImageMainTest,
+                         LoadSaveImageTestFixture);
+
+BOOST_AUTO_TEST_CASE(LoadImageTest)
+{
+  SetInputParam<vector<string>>("input", {"test_image.png", "test_image.png"});
```

Comment:
Do we need to add `test_image.png` to the repository under `src/mlpack/tests/data/`?

Reply:
It is aldeardy there I guess